src/SIM/SIMbase.h

```diff
@@ -760,9 +760,11 @@ class SIMbase : public SIMinput, public SIMdependency
   //! \param[in] nodes Total number of global nodes in model
   //! \param[in] basis The basis to specify number of DOFs for
   //! \param[in] nndof Number of DOFs on given basis
-  std::vector<int> setupAdditionalMADOF(const PatchVec& myModel, size_t nodes,
-                                        unsigned char basis,
-                                        unsigned char nndof) const;
+  //! \param[out] madof Generated MADOF array
+  void setupAdditionalMADOF(const PatchVec& myModel, size_t nodes,
+                            unsigned char basis,
+                            unsigned char nndof,
+                            std::vector<int>& madof) const;
 
```

Comment:
nei, det har du helt rett i, var litt kjapp med overgangen fra hjelpeklassen.

det blir ikke vanskelig som sådan å beregne disse up-front, men da må det til et interface for å deklarere hvilke som trengs og det fins ikke i dag.